messaging/src/main/java/org/axonframework/eventhandling/TrackingToken.java

```diff
@@ -56,4 +56,12 @@
      */
     boolean covers(TrackingToken other);
 
+    /**
+     * Return the position this token represents
+     * @return the position this token represents
+     */
+    default long position() {
```

Comment:
It's good to have a default here for Token implementations that do not (yet) support the position. However, `-1` may have a meaning for certain tokens. For stores starting at `0`, `-1` may simply mean no event have been processed yet.
Therefor, I'd recommend using an `OptionalLong` as return type. If the token doesn't support positions or can't give an estimate, it can then return `OptionalLong.empty()`.

Reply:
Done